amgut/db/patches/0040.sql

```diff
@@ -0,0 +1,220 @@
+	name                 varchar  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_extraction_robot PRIMARY KEY ( extraction_robot_id ),
+	CONSTRAINT uq_extraction_robot_name UNIQUE ( name )
+ );
+
+CREATE TABLE master_mix_lot (
+	master_mix_lot_id    bigserial  NOT NULL,
+	name                 varchar  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_master_mix_lot PRIMARY KEY ( master_mix_lot_id ),
+	CONSTRAINT uq_master_mix_lot_name UNIQUE ( name )
+ );
+
+CREATE TABLE plate_type (
+	plate_type_id        bigserial  NOT NULL,
+	name                 varchar  NOT NULL,
+	cols                 smallint  NOT NULL,
+	rows                 smallint  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_plate_type PRIMARY KEY ( plate_type_id ),
+	CONSTRAINT uq_plate_type_name UNIQUE ( name )
+ );
+
+CREATE TABLE processing_robot (
+	processing_robot_id  bigserial  NOT NULL,
+	name                 varchar  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_processing_robot PRIMARY KEY ( processing_robot_id ),
+	CONSTRAINT uq_processing_robot_name UNIQUE ( name )
+ );
+
+CREATE TABLE run (
+	run_id               bigserial  NOT NULL,
+	run_date             date  ,
+	description          varchar  ,
+	CONSTRAINT pk_run PRIMARY KEY ( run_id )
+ );
+
+CREATE TABLE sample (
+	sample_id            varchar  NOT NULL,
+	barcode              varchar  ,
+	description          varchar  ,
+	CONSTRAINT pk_sample PRIMARY KEY ( sample_id ),
+	CONSTRAINT fk_sample_barcode FOREIGN KEY ( barcode ) REFERENCES barcodes.barcode( barcode )
+ );
+CREATE INDEX idx_sample_barcode ON sample ( barcode );
+
+CREATE TABLE study (
+	study_id             bigint  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_study PRIMARY KEY ( study_id )
+ );
+COMMENT ON COLUMN study.study_id IS 'positive if Qiita study ID, negative if others';
+
+CREATE TABLE study_sample (
+	study_id             bigint  NOT NULL,
+	sample_id            varchar  NOT NULL,
+	CONSTRAINT fk_study_sample_study_id FOREIGN KEY ( study_id ) REFERENCES study( study_id )    ,
+	CONSTRAINT fk_study_sample_sample_id FOREIGN KEY ( sample_id ) REFERENCES sample( sample_id )
+ );
+CREATE INDEX idx_study_sample_study_id ON study_sample ( study_id );
+CREATE INDEX idx_study_sample_sample_id ON study_sample ( sample_id );
+
+CREATE TABLE template (
+	template_id          bigserial  NOT NULL,
+	name                 varchar  NOT NULL,
+	plate_type_id        bigint  NOT NULL,
+	description          varchar  ,
+	CONSTRAINT pk_plate_template PRIMARY KEY ( template_id ),
+	CONSTRAINT uq_template_name UNIQUE ( name ) ,
+	CONSTRAINT fk_plate_template_plate_type FOREIGN KEY ( plate_type_id ) REFERENCES plate_type( plate_type_id )
+ );
+CREATE INDEX idx_plate_template ON template ( plate_type_id );
+
+CREATE TABLE template_barcode_sequence (
+	template_id          bigint  NOT NULL,
+	col                  smallint  NOT NULL,
+	row                  smallint  NOT NULL,
```

Comment:
historically the rows have been alphabetical, would it make sense for the row to be char?